pkg/diagnose/cni.go

```diff
@@ -167,7 +163,11 @@ func checkCalicoIPPoolsIfCalicoCNI(info *cluster.Info, status reporter.Interface
 
 	checkGatewaySubnets(gateways, ippools, tracker)
 
-	return !tracker.HasFailures()
+	if tracker.HasFailures() {
+		return errors.New("Failures while diagnosing CNI")
```

Comment:
Why didn't linting flag this...

```suggestion
		return errors.New("failures while diagnosing CNI")
```